File: app/src/main/java/uk/co/islovely/ethicalshopping/MySettingsFragment.java
package uk.co.islovely.ethicalshopping;

import android.os.Bundle;
import androidx.preference.PreferenceFragmentCompat;

public class SettingsFragment : PreferenceFragmentCompat() {
    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        // Load the preferences from an XML resource
        setPreferencesFromResource(R.xml.preferences, rootKey);
    }
}

File: app/src/main/java/uk/co/islovely/ethicalshopping/MySettingsFragment.kt
package uk.co.islovely.ethicalshopping;

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat;

class MySettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences, rootKey)
    }
}
